pkg/http/sse.go

```diff
@@ -166,32 +172,25 @@ func (h sseHandler) handleEventStream(w http.ResponseWriter, r *http.Request) {
 
 		h.logger.Trace("Listening for messages", nil)
 
-		for msg := range messages {
-			msg.Ack()
-
-			response, ok := h.processMessage(w, r, msg)
-			if ok {
-				responsesChan <- response
-			}
-
+		for {
 			select {
+			case msg, ok := <-messages:
+				if !ok {
+					return
+				}
+
+				msg.Ack()
+
+				nextEvent, ok := h.streamAdapter.NextEvent(r, msg)
+				if ok {
+					h.logger.Trace("Stream responding on message", watermill.LogFields{"uuid": msg.UUID})
+					responsesChan <- nextEvent
```

Comment:
How about taking this out from the select? In case the context got canceled in the meantime we would not render the response from the adapter at least.

Reply:
How do you mean "out"? It needs to happen after we receive a message.

In the current scenario, the response won't be rendered, because `chi`'s Responder closes on context done.

Reply:
here is what I mean:
```
        var nextEvent interface{}

	select {
	case msg, ok := <-messages:
		if !ok {
			return
		}

		msg.Ack()

		e, ok := h.streamAdapter.NextEvent(r, msg)
		if ok {
			h.logger.Trace("Stream responding on message", watermill.LogFields{"uuid": msg.UUID})
		}

		nextEvent = e
	case <-r.Context().Done():
		return
	}

	if nextEvent != nil {
		responsesChan <- nextEvent
	}
```

In your implementation, in case context got cancelled during the course of h.streamAdapter.NextEvent(r, msg) execution, we put nextEvent to the rendered channel before we check for Context.Done(). If we change the implementation to somethings I suggested we would do that only if the context wasn't done.

Reply:
since this code does not change much, I am ok with your approach